src/notebooks/controllers/connectionDisplayData.ts

```diff
@@ -60,6 +62,22 @@ export class ConnectionDisplayDataProvider {
         details.category = category || details.category;
         this.details.set(connection.id, details);
 
+        if (
+            this.featuresManager.features.kernelPickerType === 'Insiders' &&
+            (connection.kind === 'connectToLiveRemoteKernel' || connection.kind === 'startUsingRemoteKernelSpec')
+        ) {
+            getRemoteServerDisplayName(connection, this.serverUriStorage)
+                .then((displayName) => {
+                    if (details.serverDisplayName !== displayName) {
+                        details.serverDisplayName = displayName;
```

Comment:
Instead of this, we can just 
```typescript
details.description = 
details.includes(displayName) ? details:`${details.description} (${displayName})`
```